### PR TITLE
HAWQ-735. Use --with-thrift to control building thrift or not & Import thrift-0.9.3.

### PR DESCRIPTION
**Pull request by xunzhang**

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @radarwave @changleicn

**Comment by xunzhang** [comment]
Usage: `./configure --with-thrift;`

**Comment by radarwave** [comment]
+1

**Comment by xunzhang** [comment]
@changleicn plz review [this file](https://github.com/xunzhang/incubator-hawq/blob/1dc1d209a22e695c34c02116b9a3e24142891372/depends/thirdparty/ImportLog).
As we discussed before, in this pr I add a file named `ImportLog` inside `depends/thirdparty` folder to record the package importing information.

**Comment by changleicn** [comment]
looks good.

**Comment by xunzhang** [comment]
Merging this into master, thanks!

File: configure [hunk 1]
```diff
@@ -1552,6 +1554,7 @@ Optional Packages:
   --with-bonjour          build with Bonjour support
   --with-openssl          build with OpenSSL support
   --without-readline      do not use GNU Readline nor BSD Libedit for editing
+  --with-thrift       do not build thrift
```
Comment (zhangh43) [inline configure]: here with-thrift confilcts with do not build

Comment (xunzhang) [inline configure]: Yes, I made a mistake, thanks.

**Pull request closed without merging**